src/app/components/media/MediaTitle.js

```diff
@@ -0,0 +1,189 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { createFragmentContainer, graphql } from 'react-relay/compat';
+import { truncateLength, emojify } from '../../helpers';
+import MediaUtil from './MediaUtil';
+
+/**
+ * The API of a <FormattedMessage> with the value of null.
+ *
+ * Usage:
+ *
+ * <NullMessage>{value => value === null ? "always" : "never"}</NullMessage>
+ */
+const NullMessage = ({ children }) => children(null);
+
+const MediaType = {
+  Twitter: 'twitter',
+  Facebok: 'facebook',
+  Instagram: 'instagram',
+  Youtube: 'youtube',
+  Tiktok: 'tiktok',
+  Claim: 'claim',
+  Image: 'image',
+  Page: 'page',
+};
+
+/**
+ * A <FormattedMessage> or <NullMessage>, depending on media type.
+ */
+const TypeMessageOrNullMessage = ({ type, children }) => {
+  switch (type) {
+  case MediaType.Twitter:
+    return <FormattedMessage id="media.typeTwitter" defaultMessage="Tweet">{children}</FormattedMessage>;
+  case MediaType.Facebook:
+    return <FormattedMessage id="media.typeFacebook" defaultMessage="Facebook post">{children}</FormattedMessage>;
+  case MediaType.Instagram:
+    return <FormattedMessage id="media.typeInstagram" defaultMessage="Instagram">{children}</FormattedMessage>;
+  case MediaType.Video:
+    return <FormattedMessage id="media.typeVideo" defaultMessage="Video">{children}</FormattedMessage>;
+  case MediaType.Tiktok:
+    return <FormattedMessage id="media.typeTiktok" defaultMessage="TikTok video">{children}</FormattedMessage>;
+  case MediaType.Claim:
+    return <FormattedMessage id="media.typeClaim" defaultMessage="Text">{children}</FormattedMessage>;
+  case MediaType.Image:
+    return <FormattedMessage id="media.typeImage" defaultMessage="Image">{children}</FormattedMessage>;
+  case MediaType.Page:
+    return <FormattedMessage id="media.typePage" defaultMessage="Page">{children}</FormattedMessage>;
+  default:
+    return <NullMessage>{children}</NullMessage>;
+  }
+};
+TypeMessageOrNullMessage.defaultProps = {
+  type: null,
+};
+TypeMessageOrNullMessage.propTypes = {
+  type: PropTypes.oneOf(Object.keys(MediaType)), // or null
+  children: PropTypes.func.isRequired,
+};
+
+function mediaTypeOrNull(projectMedia) {
+  switch (projectMedia.domain) {
+  case 'twitter.com': return MediaType.Twitter;
+  case 'facebook.com': return MediaType.Twitter;
+  case 'instagram.com': return MediaType.Twitter;
+  case 'youtube.com': return MediaType.Twitter;
+  case 'tiktok.com': return MediaType.Twitter;
```

Comment:
why all cases return `MediaType.Twitter`?

Reply:
Aaaaaah! Great catch. Thank you! Will fix.

Reply:
@caiosba Fixed. Also, I tested one of each type. And I added a new `MediaTitle.test.js` to test all cases. (It uncovered a bug -- woo-hoo!)